src/ChainSafe.GamingSDK.EVM/Web3/Core/Analytics/AnalyticsEvent.cs

```diff
@@ -41,9 +41,20 @@ public override string ToString()
             IfNotNullNorWhitespace(GasLimit, () => AppendMember(nameof(GasLimit), GasLimit));
             IfNotNullNorWhitespace(GasPrice, () => AppendMember(nameof(GasPrice), GasPrice));
 
-            foreach (var (name, value) in CustomProperties)
+            if (CustomProperties != null)
             {
-                AppendMember(name, value);
+                foreach (var pair in CustomProperties)
+                {
+                    var key = pair.Key ?? "Unnamed";
+                    var value = pair.Value ?? "Unknown";
+
+                    if (pair.Key == null || pair.Value == null)
```

Comment:
Wouldn't these values, if previously null, be assigned "Unnamed", meaning this would never trigger?

Reply:
Yes, I was wondering how to handle this. Do we want to log these errors or throw an exception?

Reply:
Yes, I'm aware of that going to remove the key/value set.